packages/database/src/models/fhir/Resource.ts

```diff
@@ -200,7 +200,20 @@ export class FhirResource extends Model {
     });
 
     for (const unresolvedResource of unresolvedResources) {
-      await this.materialiseFromUpstream(unresolvedResource.upstreamId);
+      try {
+        await this.materialiseFromUpstream(unresolvedResource.upstreamId);
+      } catch (error) {
+        if (error instanceof Error) {
+          // Rethrowing like this to preserve stacktrace while logging which resource failed to resolve
+          const rethrownError = new Error(
+            `Error resolving upstreams for ${this.fhirName}/${unresolvedResource.id}: ${error.stack ?? error.message ?? error.toString() ?? ''}`,
+          );
+          rethrownError.stack = error.stack;
+          throw rethrownError;
+        } else {
+          throw error;
+        }
```

Comment:
Apparently there's a much cleaner way of doing this as of Node 16, using the `cause` option: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

However our typescript compiler didn't seem happy with that, and I didn't want to play around with our TS config on a hotfix. I think I'll try and get it working before merging to main though :+1: